pattern/constraint/thing/ValueConstraint.java

```diff
@@ -195,6 +198,16 @@ public FunctionalIterator<AlphaEquivalence> alphaEquals(ValueConstraint<?> that)
                 .flatMap(a -> a.alphaEqualIf(this.value.equals(that.value)));
     }
 
+    public boolean maybeCompatible(ValueConstraint<?> valueConstraint) {
```

Comment:
I do wonder, when we say two predicates are "compatible", is that less clear than inverting this to say "contradicts"?

Reply:
"IsConsistentWith" ?

Reply:
well, to keep the intent we'd have to do `maybeConsistentWith` since we want to keep the `maybe`-ness about the positive check.

Reply:
if we call it `contradicts` we don't need the `maybe` because if it's not a certain contradiction it's not a contradiction. We can then also shorten the abstract method it calls to `contradictsEquality`